launch/test/launch/test_execute_process.py

```diff
@@ -173,6 +175,54 @@ def generate_launch_description():
     assert expected_called_count == on_exit_callback.called_count
 
 
+def test_execute_process_with_respawn_substitution():
+    """Test launching a process with a respawn substitution and respawn_delay attribute."""
+    def on_exit_callback(event, context):
+        on_exit_callback.called_count = on_exit_callback.called_count + 1
+
+    respawn_delay = 2.0
+    shutdown_time = 3.0  # to shutdown the launch service, so that the process only respawn once
+
+    lc = LaunchContext()
+    lc._set_asyncio_loop(asyncio.get_event_loop())
+    respawn = parse_substitution('$(var respawn)')
```

Comment:
@kosuke55 no need to parse substitutions from a string:
```suggestion
    respawn = LaunchConfiguration('respawn')
```

Reply:
Fixed in https://github.com/ros2/launch/pull/569/commits/d446ae8660059d6645ee38c5e173a1fcd74558cc.